server/midddleware/middleware.go

```diff
@@ -71,9 +70,10 @@ func Get(config *config.Config, tenantMgr *metadata.TenantManager, txMgr *transa
 		grpc_ratelimit.StreamServerInterceptor(&RateLimiter{}),
 		grpc_auth.StreamServerInterceptor(authFunction),
 		namespaceInitializer.NamespaceSetterStreamServerInterceptor(),
-		grpctrace.StreamServerInterceptor(grpctrace.WithServiceName(util.Service)),
-		grpc_logging.StreamServerInterceptor(grpc_zerolog.InterceptorLogger(sampledTaggedLogger), []grpc_logging.Option{}...),
-		validatorStreamServerInterceptor(),
+	}
+
+	if config.DatadogTrace.Enabled {
```

Comment:
Should we put this logic into the Interceptor and just pass out required components (Logger etc. Config is global so no need to pass that down)? Am I missing the benefit of having these interceptors exposed at this level?

Reply:
Initializing the interceptor has a very visible cost according to the profiler. I think it's better only to initialize them when they are needed. What do you think?